bundles/org.openhab.binding.bmwconnecteddrive/src/main/java/org/openhab/binding/bmwconnecteddrive/internal/handler/ConnectedDriveBridgeHandler.java

```diff
@@ -0,0 +1,202 @@
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.bmwconnecteddrive.internal.ConnectedDriveConfiguration;
+import org.openhab.binding.bmwconnecteddrive.internal.discovery.VehicleDiscovery;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.NetworkError;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.discovery.Dealer;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.discovery.VehiclesContainer;
+import org.openhab.binding.bmwconnecteddrive.internal.utils.Constants;
+import org.openhab.binding.bmwconnecteddrive.internal.utils.Converter;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonParseException;
+
+/**
+ * The {@link ConnectedDriveBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ConnectedDriveBridgeHandler extends BaseBridgeHandler implements StringResponseCallback {
+    private final Logger logger = LoggerFactory.getLogger(ConnectedDriveBridgeHandler.class);
+    private HttpClientFactory httpClientFactory;
+    private Optional<VehicleDiscovery> discoveryService = Optional.empty();
+    private Optional<ConnectedDriveProxy> proxy = Optional.empty();
+    private Optional<ConnectedDriveConfiguration> configuration = Optional.empty();
+    private Optional<ScheduledFuture<?>> initializerJob = Optional.empty();
+    private Optional<String> troubleshootFingerprint = Optional.empty();
+
+    public ConnectedDriveBridgeHandler(Bridge bridge, HttpClientFactory hcf) {
+        super(bridge);
+        httpClientFactory = hcf;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands available
+    }
+
+    @Override
+    public void initialize() {
+        troubleshootFingerprint = Optional.empty();
+        updateStatus(ThingStatus.UNKNOWN);
+        configuration = Optional.of(getConfigAs(ConnectedDriveConfiguration.class));
+        configuration.ifPresentOrElse(config -> {
+            proxy = Optional.of(new ConnectedDriveProxy(httpClientFactory, config));
+            // give the system some time to create all predefined Vehicles
+            initializerJob = Optional.of(scheduler.schedule(this::requestVehicles, 5, TimeUnit.SECONDS));
+        }, () -> {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        });
```

Comment:
From what I can see deep inside BaseThingHandler.getConfigAs() ... there is a call to ConfigMapper.as returning Nullable which is not handled properly and would be returend by BaseThingHandler. But I agree: As formally BaseThingHandler.getConfigAs is declared as NonNullByDefault that should be fixed within BaseThingHandler. So I agree it can be removed.

Reply:
Good catch. You could file an issue in openhab-core.

Reply:
fixed in https://github.com/weymann/openhab-addons/pull/7